src/XRP/serializer.ts

```diff
@@ -89,6 +101,15 @@ const serializer = {
       object.TxnSignature = signature
     }
 
+    Object.assign(object, this.memosToJSON(transaction.getMemosList()))
+
+    if (transaction.getMemosList() && transaction.getMemosList().length) {
+      object.Memos = transaction
+        .getMemosList()
+        .filter((memo) => !!memo)
+        .map((memo) => this.memoToJSON(memo))
+    }
```

Comment:
I don't understand this block of code. Why are we doing `Object.assign()` followed by directly setting `object.Memos`? What is actually happening in each of those two statements?

Reply:
I was doing it that way because the `Memo` is comprised of entirely optional fields, so I'm trying to reduce things to `undefined` if there's no actual data in there (i.e. someone who was using this passed in `Memo` instances that had nothing inside them). I was using `Object.assign` as a shorthand instead of getting the result, checking if it's not `undefined`, and then setting the field. 

I was also doing this due to our discussion the other day about the signing procedure that we have. I just figured reducing the potential footprint of the resulting JSON as much as possible was useful. 

Most of the work that is coming into question on this PR is a result of the way the rippled API is implemented for whatever that's worth.

Reply:
I still don't understand why you need to do both though. Like, isn't this `.getMemosList().filter().map()` call implicit in `this.memosToJSON()`?

Reply:
Haha, I just realized I accidentally left both code paths in when I committed. Sorry; no wonder this was more confusing than necessary. Pushing a fix for that shortly.